augur/filter.py

```diff
@@ -1046,8 +1046,10 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
             # Extend metadata with generated date columns
             # Drop the 'date' column since it should not be used for grouping.
             metadata = pd.concat([metadata.drop('date', axis=1), df_dates], axis=1)
-            if 'year' in generated_columns_requested:
-                # Skip ambiguous years.
+
+            # Skip ambiguous dates.
+            if True:
+                # Skip ambiguous years (always, since generated columns are requested).
```

Comment:
Purely for readability. I think this reads better than without the condition, since it follows the pattern below it and the parent condition is very far up.

```py
if generated_columns_requested:
    # ...

    # Skip ambiguous dates.
    if (some_condition):
        # Skip ambiguous years (always, since generated columns are requested).
        # <some code>
    if 'month' in generated_columns_requested or 'week' in generated_columns_requested:
        # Skip ambiguous months.
        # <some code>
    if 'week' in generated_columns_requested:
        # Skip ambiguous days.
        # <some code>
```

vs.

```py
if generated_columns_requested:
    # ...

    # Skip ambiguous dates.
    # Skip ambiguous years (always, since generated columns are requested).
    # <some code>
    if 'month' in generated_columns_requested or 'week' in generated_columns_requested:
        # Skip ambiguous months.
        # <some code>
    if 'week' in generated_columns_requested:
        # Skip ambiguous days.
        # <some code>
```

But I can change it if you prefer without.

Reply:
Hmm, if for readability only, I would prefer to just move the whole chunk for skip ambiguous dates to an external function.

Reply:
With the persistent variables, moving the whole chunk isn't the easiest option without a bigger refactor. `metadata` and `skipped_strains` are updated, while at a minimum `temp_date_cols` is required. This would look something like

```py
metadata, skipped_strains = skip_ambiguous_dates(metadata, skipped_strains, temp_date_cols)
```

which I'm not sold on. Do you know of a better way to do this?

Readability should be improved with #997 and maybe splitting functions further would make more sense following the initial splitting of `filter.py`.

Reply:
Hmm, maybe the function just checks for strains with ambiguous dates and the persistent variables can be updated outside the function:
```python

ambiguous_date_strains = get_ambiguous_date_skipped_strains(
    metadata,
    temp_prefix,
    generated_columns_requested
)
metadata.drop([record['strain'] for record in ambiguous_date_strains], inplace=True)
skipped_strains.extend(ambiguous_date_strains)
```

Reply:
Thanks!